packages/node_modules/@webex/webex-core/src/lib/services/service-host.js

```diff
@@ -0,0 +1,250 @@
+   *
+   * @param {SetStatusPTO} pto - The parameter transfer object.
+   * @returns {this}
+   */
+  setStatus({failed, replaced}) {
+    if (failed || failed === false) {
+      this.failed = failed;
+    }
+
+    if (replaced || replaced === false) {
+      this.replaced = replaced;
+    }
+
+    return this;
+  }
+
+  /**
+   * The parameter transfer object for {@link polyGenerate}.
+   *
+   * @typedef {object} PolyGeneratePTO
+   * @property {object} PolyGeneratePTO.catalog - The target catalog.
+   * @property {string} PolyGeneratePTO.name - The service name.
+   * @property {string} PolyGeneratePTO.url - The service url.
+   */
+
+  /**
+   * Generate a service host using only a catalog, name, and URL.
+   *
+   * @param {PolyGeneratePTO} pto - The parameter transfer object.
+   * @returns {ServiceHost} - The generated service host.
+   */
+  static polyGenerate({catalog, name, url}) {
+    return new ServiceHost({
+      catalog,
+      defaultUri: url,
+      hostGroup: Url.parse(url).host,
+      id: (name) ? `poly-head:poly-group:poly-cluster:${name}` : undefined,
+      priority: 1,
+      uri: Url.parse(url).host
+    });
+  }
+
+  /**
+   * Validate that a constructor parameter transfer object is valid.
+   *
+   * @param {ConstructorPTO} pto - The parameter transfer object.
+   * @throws - If the parameter transfer object is not valid.
+   * @returns {void}
+   */
+  static validate(pto) {
+    // Destructure the parameter transfer object.
+    const {
+      catalog,
+      defaultUri,
+      hostGroup,
+      id,
+      priority,
+      uri
+    } = pto;
+
+    // Generate the error message.
+    let message;
+    let foundError;
+
+    // Validate the catalog property.
+    if (!catalog || !SERVICE_CATALOGS.includes(catalog)) {
+      message += '\'catalog\' must be a string';
+      foundError = true;
+    }
+
+    // Validate the `defaultUri` property.
+    if (!defaultUri || typeof defaultUri !== 'string') {
+      message += '\'defaultUri\' must be a string';
+      foundError = true;
+    }
+
+    // Validate the `hostGroup` property.
+    if (!hostGroup || typeof hostGroup !== 'string') {
+      message += '\'hostGroup\' must be a string';
+      foundError = true;
+    }
+
+    // Validate the `id` property.
+    if (!id || typeof id !== 'string' || id.split(':').length !== 4) {
+      message += '\'id\' must be a string that contains 3 \':\' characters';
+      foundError = true;
+    }
+
+    // Validate the `priority` property.
+    if (!priority || typeof priority !== 'number') {
+      message += '\'priority\' must be a number';
+      foundError = true;
+    }
+
+    // Validate the `uri` property.
+    if (!uri || typeof uri !== 'string') {
+      message += '\'uri\' must be a string';
+      foundError = true;
+    }
+
+    // Validate that an error was found.
+    if (foundError) {
+      throw new Error(
+        `service-host: invalid constructor parameters, ${message}`
+      );
+    }
```

Comment:
Instead of this, you an create a method inside of this method that looks like this:
`` const throwError = (m) => throw new Error(`service-host: invalid constructor parameters, ${m}`);``
and replace all of the message assignments and `foundError = true;` to `throwError('Whatever the error message is');`

Reply:
Great idea, I'll re-write this way.